src/main.js

```diff
@@ -0,0 +1,269 @@
+            showPage('playback-page')
+        });
+    }
+
+    function renderPlaybackPage() {
+        videoController.startVideo(currentVideoStream);
+
+        setFocus(videoController.video, null, action => {
+            if (action == inputActions.select || action == inputActions.playPause) {
+                videoController.togglePlayPause();
+                return true; // handled
+            }
+
+            if (action == inputActions.fastForward || action == inputActions.moveRight
+                || action == inputActions.rightShoulder1 || action == inputActions.rightShoulder2) {
+                videoController.stepForward();
+                return true; // handled
+            }
+
+            if (action == inputActions.rewind || action == inputActions.moveLeft
+                || action == inputActions.leftShoulder1 || action == inputActions.leftShoulder2) {
+                videoController.stepBackward();
+                return true; // handled
+            }
+
+            if (action == inputActions.num2 || action == inputActions.rightStick) {
+                // QA helper to allow ads to be skipped.
+                videoController.skipAd();
+                return true; // handled
+            }
+        });
+    }
+
+    function newFocusable(elementRef, selectAction, inputAction) {
+        return new Focusable(elementRef, selectAction, inputAction, focusManager);
+    }
+
+    function setFocus(elementRef, selectAction, inputAction) {
+        focusManager.setContentFocusables([newFocusable(elementRef, selectAction, inputAction)]);
+    }
+
+    function onBackAction(event) {
+        // Since the true[X] ad renderer also needs to field this event, we need to ignore when the user
+        // backs out of the ad overlay.
+        //
+        // We do this by only recognizing a back action to this app's specific state.
+        const state = event && event.state;
+        const isForThisApp = state && state.app == config.name && state.isBlock;
+        if (!isForThisApp) return; // let the back action proceed, most likely from ad overlay processing.
+
+        pushBackActionStub(); // ensure the next back action for this app is blocked.
+
+        returnToParentPage();
+    }
+
+    function pushBackActionBlock() {
+        history.pushState({app: config.name, isBlock: true}, null, null);
+        pushBackActionStub(); // push a history state that can be consumed for this app's back action.
+    }
+
+    function pushBackActionStub() {
+        history.pushState({app: config.name, isStub: true}, null, null);
+    }
+
+    function returnToParentPage() {
+        let returnToPage = 'home-page';
+        if (currentPage == 'debug-log') {
+            returnToPage = lastPage;
+        }
+        showPage(returnToPage);
+    }
+
+    function initializeApplication() {
+        try {
+            console.log(`running ${config.name} ${config.version} ${config.buildDate}
```

Comment:
Would be good to also console.log the specific version of TAR that is running. Perhaps it's more appropriate in the interactive ad js file. But this would help QA identify more specifically the combined version of ref app + TAR that is at issue with a given problem.

Reply:
we can do this, we have tar.version already present for this purpose